### ISSUE TITLE
Console: 'table' API

### ISSUE THREAD

**Issue by bryphe** [facebookexperimental/reason-native#27]
Talking to @jordwalke on Discord - it sounds like there is a `console.table` API planned, so I wanted to open an issue to discuss. 

This would be very useful for formatting benchmarks, performance tests, etc.

The API would need to differ from the Node-style API: https://nodejs.org/api/console.html#console_console_table_tabulardata_properties , as that API depends on record keys, which wouldn't be available for OCaml. 

Has there already been thinking on what the API could look like for this?

**Comment by kyldvs** [comment]
Yeah, I've already made a lot of progress. These are the passing test cases I have right now, I just need to clean it up and get this released. Let me know what you think of the API, right now it uses JSX.

https://gist.github.com/kyldvs/22f5d85e287cd006dcdd89f64c1c4b39

**Comment by bryphe** [comment]
@kyldvs - API looks great to me! Thanks for sharing out the progress so far.

**Comment by kyldvs** [comment]
No problem! The main blocker right now is something @bandersongit pointed out.

We want to allow printing things in color in the table, so in order to pad cells and wrap correctly we have to accept a custom length function (to ignore the terminal color codes when computing length). But right now we don't have a way for the user to say where it is safe to break for cell wrapping, so the colors could bleed over cells unless we factor that out somehow.
